pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,368 @@
+        Distance from source to lenses. This is about 160 m at XPP.
+    material: str
+        Beryllium. The use of beryllium extends the range of operation
+        of compound refractive lenses, improving transmission,
+        aperture size, and gain
+    density: TODO: find out what density is
+    fwhm_unfocused: float
+        This is about 400 microns at XPP.
+    printsummary: boolean
+        Prints summary of parameters/calculations if True
+
+    Returns
+    -------
+    float: Size FWHM
+    """
+    # Focal length for certain lenses configuration and energy
+    focal_length = calc_focal_length(E, lens_set, material, density)
+    
+    # use lens makers equation to find distance to image of source
+    if source_distance is not None:
+        focal_length = 1 / (1 / focal_length - 1 / source_distance)
+    
+    lam = 1.2398 / E * 1e-9
+    # the w parameter used in the usual formula is 2*sigma
+    w_unfocused = fwhm_unfocused * 2 / 2.35
+    # assuming gaussian beam divergence = w_unfocused/f we can obtain
+    waist = lam / np.pi * focal_length / w_unfocused
+    rayleigh_range = np.pi * waist ** 2 / lam
+    size = waist * np.sqrt(1.0 + (distance - focal_length) ** 2.0 / rayleigh_range ** 2)
+    size_fwhm = size * 2.35 / 2.0
+
+    if printsummary:
+        print("FWHM at lens   : %.3e" % (fwhm_unfocused))
+        print("waist          : %.3e" % (waist))
+        print("waist FWHM     : %.3e" % (waist * 2.35 / 2.0))
+        print("rayleigh_range : %.3e" % (rayleigh_range))
+        print("focal length   : %.3e" % (focal_length))
+        print("size           : %.3e" % (size))
+        print("size FWHM      : %.3e" % (size_fwhm))
+
+    return size_fwhm
+
+
+def find_energy(lens_set, distance=3.952, material="Be", density=None):
+    """
+    Finds the energy that would focus at a given distance (default = 4m)
+
+    Parameters
+    -----------
+    lens_set : `list`
+        [(numer1, lensthick1), (number2, lensthick2)...]
+    distance : `float`
+    material: str
+        Beryllium. The use of beryllium extends the range of operation
+        of compound refractive lenses, improving transmission,
+        aperture size, and gain
+    density: TODO: find out what density is
+    
+    usage findEnergy( (2,200e-6,4,500e-6) ,distance =4 )
+  """
+    Emin = 1.0
+    Emax = 24.0
+    E = (Emax + Emin) / 2.0
+    absdiff = 100
+    while absdiff > 0.0001:
+        fmin = calc_focal_length(Emin, lens_set, material, density)
+        fmax = calc_focal_length(Emax, lens_set, material, density)
+        E = (Emax + Emin) / 2.0
+        f = calc_focal_length(E, lens_set, material, density)
+        if (distance < fmax) and (distance > f):
+            Emin = E
+        elif (distance > fmin) and (distance < f):
+            Emax = E
+        else:
+            print("somehow failed ...")
+            break
+        absdiff = abs(distance - f)
+    print("Energy that would focus at a distance of %.3f is %.3f" % (distance, E))
```

Comment:
We should also try to enforce `print` -> `logger` changes, I think.

Reply:
Ok that was something I was not sure about, i'll switch to using `logger`